### ISSUE TITLE
posixlib `errno.scala` is incomplete

### ISSUE THREAD

**Issue by LeeTibbert** [scala-native/scala-native#2716]
1) According to the POSIX 2018 specification for `errno.h`, the following ought to 
     provide access to the `errno` variable:
     ```
    import scala.scalanative.posix.{errno => posixErrno, posixErrno.errno}
    ```

    [I will note and skip in passing the long standing flaw of the object having been named (lower case) `errno`,
    causing conflict with the enclosed variable. Fi!]

    Currently, in order to work in POSIX land, and use both errno and POSIX defined
    values for errno, one must import the variable `errno` from clib and then the
    values from posixib.  The latter is comprehensible, the former is a case where
    we can and should do better.
    ```
     // Taken directly from unit-tests TimeTest.scala
     // Declare as:
     import scalanative.libc.{errno => libcErrno}
     import scala.scalanative.posix.errno.{EINVAL, EINTR}

    // Use as:
    assertEquals(
        s"clock_nanosleep failed with errno: ${libcErrno.errno}",
        EINTR,
        libcErrno.errno
      )
     ```

2) The list of possible values for errno should be checked against the specification.
    I know of no missing value but believe that validation would be worthwhile,
    if only bringing (my) peace of mind.

    Later: posixlib `errno.scala` was compared to the 2018 POSIX specification
              and the symbol EADDRNOTAVAIL was found to be missing in 
              the former. See Issue #2717.

**Comment by LeeTibbert** [comment]
I have posixlib errno variable working in my private area and hope to submit it after
PR #2718 is merged.  That will keep the changes associated with the two different
concepts separated and ease PR submission & review.

**Comment by LeeTibbert** [comment]
resolved by merged PR #2721
